cli/cmd/agent_list.go

```diff
@@ -0,0 +1,208 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2022, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cmd
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/lacework/go-sdk/api"
+)
+
+var (
+	agentListCmdState = struct {
+		// The available filters for the agent list command
+		AvailableFilters CmdFilters
+
+		// List of filters to apply to the agent list command
+		Filters []string
+	}{}
+
+	agentListCmd = &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "List all hosts with a running agent",
+		Long: `List all hosts that have a running agent in your environment.
+
+You can use 'key:value' pairs to filter the list of hosts with the --filter flag.
```

Comment:
Reading this description, it sounds like i should not be able to pass in a value without a `:` however this runs without errpr

```
❯ lacework agent list --filter invalid
- Fetching list of agents with filters (invalid)... 
```